### PR TITLE
Added audio Safeguards for sliderbars

### PR DESCRIPTION
**Pull request by ghost**
This checks if the slider is at the minimum or maximum value before playing any audio.  It prevents playing any audio while holding the mouse button down and moving the slider further than the minimum or maximum values.

I'm Tayo in the Discord by the way.

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuSliderBar.cs [hunk 1]
```diff
@@ -66,11 +66,15 @@ private void load(AudioManager audio, OsuColour colours)
...
         private void playSample()
         {
-            if (Clock == null || Clock.CurrentTime - lastSampleTime <= 50)
-                return;
-            lastSampleTime = Clock.CurrentTime;
-            sample.Frequency.Value = 1 + NormalizedValue * 0.2f;
-            sample.Play();
+            if (MathHelper.Clamp(
+                nub.DrawPosition.X - nub.DrawWidth / 2, 0, DrawWidth) > 0 & MathHelper.Clamp(
+                DrawWidth - nub.DrawPosition.X - nub.DrawWidth / 2, 0, DrawWidth) > 0) {
```
Comment (smoogipooo) [inline osu.Game/Graphics/UserInterface/OsuSliderBar.cs]: This is very messy and should not be necessary. Update Framework + merge origin/master into this branch, then do playSample() inside UpdateValue.

**Comment by smoogipooo** [comment]
Gotta remove the OnClick and OnKeyDown ones too.

**Comment by peppy** [comment]
i don't even think this is the behaviour we want tbh

**Comment by lesleyxxx12** [comment]
Can you elaborate?

**Comment by peppy** [comment]
i like it playing sounds when at the end

though it should probably freq shift as per osu-stable behaviour

**Comment by lesleyxxx12** [comment]
Thing is though, when the slider was at the max value audio would still play when the mouse was held down. That's what I tried to fix in the first place.

**Comment by peppy** [comment]
which i'm saying is probably fine and intentional. i'll check and confirm when i'm at a pc

**Comment by ColdVolcano** [comment]
I don't think it's intentional https://streamable.com/uzxq2 but up to ppy

**Comment by lesleyxxx12** [comment]
Exactly that. Thanks for the comparison.

**Comment by peppy** [comment]
see #622

**Pull request closed without merging**